src/app/queries/hosts.ts

```diff
@@ -0,0 +1,35 @@
+import { QueryResult } from 'react-query';
+import { usePollingContext } from '@app/common/context';
+import { POLLING_INTERVAL } from './constants';
+import { useMockableQuery, getApiUrl } from './helpers';
+import { MOCK_HOSTS } from './mocks/hosts.mock';
+import { IHost } from './types';
+import { useProvidersQuery } from '.';
+
+export const HOSTS_QUERY_KEY = 'hosts';
+
+// TODO handle error messages? (query.status will correctly show 'error', but error messages aren't collected)
+export const useHostsQuery = (providerName?: string): QueryResult<IHost[]> => {
+  const { isPollingEnabled } = usePollingContext();
+  const { data: providers } = useProvidersQuery();
+  const currentProvider = providers?.vsphere.find((provider) => provider.name === providerName);
+
+  const result = useMockableQuery<IHost[]>(
+    {
+      queryKey: providers && HOSTS_QUERY_KEY,
+      queryFn: () =>
+        fetch(getApiUrl(`${currentProvider?.selfLink || ''}/hosts?detail=true`)).then((res) =>
+          res.json()
+        ),
+      config: {
+        enabled: !!currentProvider,
+        refetchInterval: isPollingEnabled ? POLLING_INTERVAL : false,
+      },
+    },
+    MOCK_HOSTS
+  );
+  return {
+    ...result,
+    data: result.data,
+  };
```

Comment:
@ibolton336 sorry to bug you with one last change but just in case, can we return `sortResultsByName(result)` here? (imported from `@app/queries/helpers`)? Considering the order of API results doesn't seem stable we need to sort all of our query data I think.

Reply:
No prob

Reply:
@mturley Should be ready now

Reply:
Haha I saw the `sortIndexedResultsByName`.. maybe those helper names are confusing

Reply:
I am ok with the names. I was just caught being lazy copying what you had for providers.